### PR TITLE
Update Go to 1.16

### PR DESCRIPTION
**Pull request by yannh**

### CODE REVIEW

**Comment by yannh** [comment]
Seems related to https://blog.golang.org/go116-module-changes
The bad news is, it looks like we might not have been pinning our deps correctly. We build coredns after copying our files in the library folder, and the build step automatically downloaded dependencies in the past - disregarding our go.mod. I guess it is good this broke? :]

**Comment by invidian** [comment]
I'm not sure if that's the right solution, but something like this works:
```patch
$ git diff
diff --git Dockerfile Dockerfile
index 9c7e5b0..6a643aa 100644
--- Dockerfile
+++ Dockerfile
@@ -1,4 +1,4 @@
-FROM golang:1.15-buster AS builder
+FROM golang:1.16-buster AS builder

 RUN apt update && apt upgrade -y && apt install iptables -y

@@ -13,6 +13,8 @@ RUN mkdir -p plugin/nodecache
 RUN echo 'nodecache:nodecache' >> /coredns/plugin.cfg

 COPY *.go /coredns/plugin/nodecache/
+RUN go get github.com/coreos/go-iptables@v0.4.5 github.com/vishvananda/netlink@v1.1.0
+
 RUN make
 RUN chmod 0755 /coredns/coredns

```

**Review by elblivion — CHANGES_REQUESTED** [review]
I think we're doing modules wrong here - this worked for me locally:
https://github.com/contentful-labs/coredns-nodecache/compare/go-1.16-modules?expand=1

**Review by elblivion — APPROVED** [review]
I get this now, thanks @invidian https://github.com/contentful-labs/coredns-nodecache/commit/d15588a80d2503aa9d79d4e6a9d82debdd7ede18#r51559385

TBH this repo is kind of confusing then, shouldn't we remove go.mod and go.sum from our repo?

File: Dockerfile [hunk 1]
```diff
@@ -13,6 +13,7 @@ RUN mkdir -p plugin/nodecache
 RUN echo 'nodecache:nodecache' >> /coredns/plugin.cfg
 
 COPY *.go /coredns/plugin/nodecache/
+RUN go get github.com/coreos/go-iptables@f901d6c2a4f2a4df092b98c33366dfba1f93d7a0 github.com/vishvananda/netlink@f049be6f391489d3f374498fe0c8df8449258372
```
Comment (invidian) [inline Dockerfile:16]: Why use commit hash rather than tags?

Comment (yannh) [reply Dockerfile:16]: What a tag points to can change, go.mod  usually has a go.sum to ensure two builds build the same.. so a weak try for more reproducibility. It's not less readable though, go.mod/sum would be neater :/

Comment (invidian) [reply Dockerfile:16]: Well, usually tags are treated as immutable and they are definitely well more maintainable than commit hashes (with no corresponding tag comment or CI to check if it matches go.mod file in this repository), but fair enough :)

**Pull request merged**